api/src/main/java/org/openmrs/ui/framework/page/PageContext.java

```diff
@@ -189,7 +190,14 @@ public void setLocale(Locale locale) {
 
     @Override
     public void includeResource(Resource resource) {
-        resourcesToInclude.add(resource);
+    	if (resourcesToInclude != null) {
+    		for (Resource res : resourcesToInclude) {
+    			if (res.getProviderName().equals(resource.getProviderName()) && res.getResourcePath().equals(resource.getResourcePath())) {
+    				throw new UiFrameworkException("Resource '"+ resource.getResourcePath() +"' Already Included in PageContext");

```

Comment:
Are you sure throwing this exception will not result into failure to run the rest of the application?

Reply:
Alright, let me look at it 🙂